src/java/org/apache/cassandra/db/tries/InMemoryDeletionAwareTrie.java

```diff
@@ -0,0 +1,459 @@
+    /// not provided with the path to the modified markers. The transformer can return null if deletions cancel out
+    /// or should not be preserved.
+    /// @param existingDeleter a function used to apply a deletion marker to potentially delete live data. This is
+    /// only called if there is both content and deletion at a given covered point. It should return null if the entry
+    /// is to be deleted.
+    /// @param insertedDeleter a function used to filter incoming entries that are covered by existing deletions
+    /// in this trie, called only if both an entry and a deletion apply to a given point. This function is not provided
+    /// with a path to the modified data.
+    /// @param deletionsAtFixedPoints True if deletion branches are at predetermined positions.
+    /// @see DeletionAwareTrie.MergeResolver#deletionsAtFixedPoints
+    public <V, E extends RangeState<E>>
+    void apply(DeletionAwareTrie<V, E> mutation,
+               final UpsertTransformer<T, V> dataTransformer,
+               final UpsertTransformer<D, E> deletionTransformer,
+               final UpsertTransformer<T, E> existingDeleter,
+               final BiFunction<D, V, V> insertedDeleter,
+               boolean deletionsAtFixedPoints,
+               Predicate<NodeFeatures<V>> needsForcedCopy)
+            throws TrieSpaceExhaustedException
+    {
+        apply(mutation,
+                (UpsertTransformerWithKeyProducer<T, V>) dataTransformer,
+                deletionTransformer, existingDeleter, insertedDeleter, deletionsAtFixedPoints, needsForcedCopy);
+    }
+
+    class DumpCursor extends InMemoryReadTrie<T>.DumpCursor<DeletionAwareInMemoryCursor<T, D>> implements DeletionAwareCursor<String, D>
+    {
+        DumpCursor(DeletionAwareInMemoryCursor<T, D> source, Function<T, String> contentToString)
+        {
+            super(source, contentToString);
+        }
+
+
+        @Override
+        public RangeCursor<D> deletionBranchCursor(Direction direction)
+        {
+            return source.deletionBranchCursor(direction);
+        }
+
+        @Override
+        public DumpCursor tailCursor(Direction direction)
+        {
+            throw new AssertionError();
+        }
+    }
+
+    public String dump(Function<T, String> contentToString)
+    {
+        return dump(contentToString, Object::toString);
+    }
+
+    /// Override of dump to provide more detailed printout that includes the type of each node in the trie.
+    /// We do this via a wrapping cursor that returns a content string for the type of node for every node we return.
+    public String dump(Function<T, String> contentToString, Function<D, String> rangeToString)
+    {
+        return new DumpCursor(makeCursor(Direction.FORWARD), contentToString).process(new TrieDumper.DeletionAware<>(Function.identity(), rangeToString));
+    }
+
+    private String dumpBranch(int branchRoot)
```

Comment:
Seems unused

Reply:
Suppressed warning and added explanation.